src/gui/inputdevices/TouchInputHandler.h

```diff
@@ -11,23 +11,31 @@
 
 #pragma once
 
-#include <XournalType.h>
+#include "../../util/XournalType.h"
 #include "AbstractInputHandler.h"
+#include <cmath>
 
 class InputContext;
 
 class TouchInputHandler : public AbstractInputHandler
 {
 private:
 	XOJ_TYPE_ATTRIB;
-	GdkEventSequence* currentSequence = nullptr;
-	double lastPosX = -1.0;
-	double lastPosY = -1.0;
+	GdkEventSequence* primarySequence = nullptr;
+  GdkEventSequence* secondarySequence = nullptr;
+	double priLastPosX = -1.0;
+	double priLastPosY = -1.0;
+  double secLastPosX = -1.0;
+  double secLastPosY = -1.0;
+  double lastDiff = 0.0;
 
 private:
 	void actionStart(InputEvent* event);
 	void actionMotion(InputEvent* event);
 	void actionEnd(InputEvent* event);
```

Comment:
Rename those to `scroll...` to make their purpose clear as we have two different functionalities in this class now.